packages/core/src/common/promise-util.ts

```diff
@@ -27,3 +27,20 @@ export class Deferred<T> {
         this.reject = reject;
     });
 }
+
+/**
+ * Simple timeout-as-promise implementation, can be cancelled.
+ *
+ * @param time delay to wait for
+ * @param value value to resolve to after the delay
+ */
+export function delay<T>(time: Number, value?: T): CancellablePromise<T> {
```

Comment:
why is it needed? maybe you need something like `lodash.debounce` or `lodash.throttle`?

Reply:
No, it was for a timeout as a promise, just added the extra sugar of being able to cancel it, but it is basically a `setTimeout` that works as a promise.

Reply:
ok, so it could be as simple as: `new Promise(resolve => setTimeout(delay, resolve))`?

I would need to look at code closer to undestand what kind of optimizaition you are trying to achieve. I don't really understand yet why polling is needed.

Reply:
The polling is for the user, so that you can see which server is online and which is not.

Be it while typing or looking through the history.

Basically I dispatch a simple GET request to addresses in the history and for the typed URL. The timeout is here so that we can hop out of the waiting for every request done concurrently. The problem is that by default the request timeout can be very long, so it could block a `Promise.all` call.

Instead I run everything and wait a certain amount of time.

Reply:
ok, simple `timeout` should be fine, please don't overdo with utilities, I would prefer if developers rely on standard APIs than our sugar, i.e. `CancellablePromise` seems to be already over the top and not used anywhere